src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/LdapPal.Linux.cs

```diff
@@ -92,12 +92,26 @@ internal static int ParseResultReferral(ConnectionHandle ldapHandle, IntPtr resu
 
         internal static int SetPtrOption(ConnectionHandle ldapHandle, LdapOption option, ref IntPtr inValue) => Interop.Ldap.ldap_set_option_ptr(ldapHandle, option, ref inValue);
 
+        internal static int SetStringOption(ConnectionHandle ldapHandle, LdapOption option, string inValue) => Interop.Ldap.ldap_set_option_string(ldapHandle, option, inValue);
+
         internal static int SetReferralOption(ConnectionHandle ldapHandle, LdapOption option, ref LdapReferralCallback outValue) => Interop.Ldap.ldap_set_option_referral(ldapHandle, option, ref outValue);
 
         // This option is not supported in Linux, so it would most likely throw.
         internal static int SetServerCertOption(ConnectionHandle ldapHandle, LdapOption option, VERIFYSERVERCERT outValue) => Interop.Ldap.ldap_set_option_servercert(ldapHandle, option, outValue);
 
-        internal static int BindToDirectory(ConnectionHandle ld, string who, string passwd) => Interop.Ldap.ldap_simple_bind(ld, who, passwd);
+        internal static int BindToDirectory(ConnectionHandle ld, string who, string passwd)
+        {
+            IntPtr passwordPtr = IntPtr.Zero;
+            try
+            {
+                berval passwordBerval = new berval { bv_len = passwd.Length, bv_val = LdapPal.StringToPtr(passwd) };
+                return  Interop.Ldap.ldap_sasl_bind(ld, who, null, passwordBerval, IntPtr.Zero, IntPtr.Zero, IntPtr.Zero);
+            }
+            finally
+            {
+                Marshal.FreeHGlobal(passwordPtr);
```

Comment:
I must be missing something, but where is this used?

Reply:
Not at all. It's probably supposed to be used to store this ptr: https://github.com/dotnet/runtime/blob/0fc56b96739bc378a73132932028fb0614d0c970/src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/LdapPal.Linux.cs#L107

Reply:
There seem to be two similar occurences of the code, one of which is looks correct:
- https://github.com/dotnet/runtime/blob/0fc56b96739bc378a73132932028fb0614d0c970/src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Linux.cs#L88-L103
- https://github.com/dotnet/runtime/blob/0fc56b96739bc378a73132932028fb0614d0c970/src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/Interop/LdapPal.Linux.cs#L104-L113

Maybe it could be interesting to deduplicate this?

Reply:
I'm fine choosing to dedupe or not, I'll leave that to @iinuwa but at least we should fix the occurrence when we are not storing the pointer for the berval password.

Reply:
Oh, sorry for the oversight. I went ahead and combined them since they're doing the same thing now.